src/entity/Post.ts

```diff
@@ -179,3 +184,238 @@ export const getAuthorPostStats = (
     .from(Post, 'post')
     .where({ authorId })
     .getRawOne<PostStats>();
+
+export interface AddPostData {
+  id: string;
+  title: string;
+  url: string;
+  publicationId: string;
+  publishedAt?: string | Date;
+  createdAt?: Date;
+  image?: string;
+  ratio?: number;
+  placeholder?: string;
+  tags?: string[];
+  siteTwitter?: string;
+  creatorTwitter?: string;
+  readTime?: number;
+  canonicalUrl?: string;
+  keywords?: string[];
+  description?: string;
+  toc?: Toc;
+  summary?: string;
+}
+
+const parseReadTime = (
+  readTime: number | string | undefined,
+): number | undefined => {
+  if (!readTime) {
+    return undefined;
+  }
+  if (typeof readTime == 'number') {
+    return Math.floor(readTime);
+  }
+  return Math.floor(parseInt(readTime));
+};
+
+type Reason = 'exists' | 'author banned' | 'missing fields';
+export type AddNewPostResult =
+  | { status: 'ok'; postId: string }
+  | { status: 'failed'; reason: Reason; error?: Error };
+
+const checkRequiredFields = (data: AddPostData): boolean => {
+  return !!(data && data.title && data.url && data.publicationId);
+};
+
+const shouldAddNewPost = async (
+  con: Connection,
+  data: AddPostData,
+): Promise<Reason | null> => {
+  const p = await con
+    .getRepository(Post)
+    .createQueryBuilder()
+    .select('id')
+    .where(
+      'url = :url or url = :canonicalUrl or "canonicalUrl" = :url or "canonicalUrl" = :canonicalUrl',
+      { url: data.url, canonicalUrl: data.canonicalUrl },
+    )
+    .getRawOne();
+  if (p) {
+    return 'exists';
+  }
+  if (data.creatorTwitter === '@NewGenDeveloper') {
+    return 'author banned';
+  }
```

Comment:
Is this now a weird way of checking for banned authors? 
Perhaps we could keep track of this via an array definition so it will be easier to maintain in the future?

Reply:
This was a temporary fix for someone who kept spamming our feed with low quality.
But I agree for now I moved it into an array and later we will move it to a database